lib/brain.js

```diff
@@ -298,18 +298,27 @@ Brain.prototype.checkWifiStatus = function checkWifiStatus () {
   this._transitionState('booting')
 
   this.wifi.status(function (err, status, ip) {
-    if (err || status === 'bill') {
-      if (err) console.log(err.stack)
-      if (self.state !== 'wifiConnecting') self._wifiConnecting()
-      self.wifi.waitConnection(function (err, ip) {
-        if (err) {
-          self.wifi.startScanning()
-          self._wifiList()
-          return
-        }
-        self.config.ip = ip
-        self._wifiConnected()
-      })
+    if (err) {
+      if (err)
+        console.log(err.stack)
+      if (deviceConfig.cryptomatModel === 'Douro I') {
+        if (self.state !== 'wifiConnecting')
+          self._wifiConnecting()
+
+        self.wifi.waitConnection(function (err, ip) {
+          if (err) {
+            self.wifi.startScanning()
+            self._wifiList()
+            return
+          }
+          self.config.ip = ip
+          self._wifiConnected()
+        })
+      }
+      else {
+        self.initTrader()
```

Comment:
Note that the communication loop handled by the Trader module is what ends up presenting the 'Network Down' screen when too much negotiations with the server fail sequentially. It is also handles the recovery of the connection when it is lost.